x/wasmplus/keeper/querier_test.go

```diff
@@ -28,13 +31,81 @@ func TestQueryInactiveContracts(t *testing.T) {
 	require.NoError(t, err)
 
 	q := Querier(keeper)
-	rq := types.QueryInactiveContractsRequest{}
-	res, err := q.InactiveContracts(sdk.WrapSDKContext(ctx), &rq)
-	require.NoError(t, err)
-	expect := []string{example1.Contract.String(), example2.Contract.String()}
-	for _, exp := range expect {
-		assert.Contains(t, res.Addresses, exp)
+	specs := map[string]struct {
+		srcQuery           *types.QueryInactiveContractsRequest
+		expAddrs           []string
+		expPaginationTotal uint64
+		expErr             error
+	}{
+		"req nil": {
+			srcQuery: nil,
+			expErr:   status.Error(codes.InvalidArgument, "empty request"),
+		},
+		"query all": {
+			srcQuery:           &types.QueryInactiveContractsRequest{},
+			expAddrs:           []string{example1.Contract.String(), example2.Contract.String()},
+			expPaginationTotal: 2,
+		},
+		"with pagination offset": {
+			srcQuery: &types.QueryInactiveContractsRequest{
+				Pagination: &query.PageRequest{
+					Offset: 1,
+				},
+			},
+			expAddrs:           []string{example1.Contract.String()},
+			expPaginationTotal: 2,
+		},
+		"with invalid pagination key": {
+			srcQuery: &types.QueryInactiveContractsRequest{
+				Pagination: &query.PageRequest{
+					Offset: 1,
+					Key:    []byte("test"),
+				},
+			},
+			expErr: fmt.Errorf("invalid request, either offset or key is expected, got both"),
+		},
+		"with pagination limit": {
+			srcQuery: &types.QueryInactiveContractsRequest{
+				Pagination: &query.PageRequest{
+					Limit: 1,
+				},
+			},
+			expAddrs:           []string{example2.Contract.String()},
+			expPaginationTotal: 0,
+		},
+		"with pagination next key": {
+			srcQuery: &types.QueryInactiveContractsRequest{
+				Pagination: &query.PageRequest{
+					Key: fromBase64("AAAAAAAAAAM="),
+				},
+			},
+			expAddrs:           []string{},
```

Comment:
I think we should check for valid keys rather than invalid ones, as that time the expAddrs should be []string{example2.Contract.String()}